src/connectionController.ts

```diff
@@ -204,19 +225,233 @@ export default class ConnectionController {
     }
   }
 
+  _notifyKeytarMigrationError(migrationError: Error) {
+    const notificationAction = {
+      title: "Don't show again",
```

Comment:
How do we know that it won't be shown again? The closing information message just dismisses it. Users often ask for a feature to disable some notifications and this message can be misleading that we start to support closing messages forever.

Reply:
But why it will appear again and again? Shouldn't the migration happen only once?

Reply:
There is one (more like theoretical) case which is if the migration failed for an error which is not `KeytarModuleUnavailableError`. I referred that [here](https://github.com/mongodb-js/vscode/pull/550/files#diff-c2dbb92c81a472b4bc81088731816dd9f4676cdb1b4a8ade3fdb49c6a3b20d82R411-R421)

Reply:
I have a feeling that this is the kind of project that needs a design doc.